row_transform/row_transform.tex

```diff
@@ -0,0 +1,188 @@
+    hold, the transformation returned by \texttt{make\_row\_by\_row} has the following properties:
+    \begin{enumerate}
+        \item \textup{(Appropriate output domain).} For every element $v$ in \texttt{input\_domain}, $\function(v)$ is in \texttt{output\_domain}. 
+        
+        \item \textup{(Domain-metric compatibility).} The domain \texttt{input\_domain} matches one of the possible domains listed in the definition of \texttt{input\_metric}, and likewise \texttt{output\_domain} matches one of the possible domains listed in the definition of \texttt{output\_metric}.
+        
+        \item \textup{(Stability guarantee).} For every pair of elements $v, w$ in \texttt{input\_domain} and for every pair $(\din, \dout)$, where $\din$ is of the associated type for \texttt{input\_metric} and $\dout$ is the associated type for \texttt{output\_metric}, if $v,w$ are $\din$-close under \texttt{input\_metric} and $\Relation(\din, \dout) = \True$, then $\function(v), \function(w)$ are $\dout$-close under \texttt{output\_metric}.
+    \end{enumerate}
+\end{theorem}
+\begin{proof}
+\begin{enumerate}
+\item \textbf{(Appropriate output domain).} In the case of \texttt{make\_row\_by\_row}, this corresponds to showing that for every vector $v$ of elements of type \texttt{DIA}, \texttt{function}$(v)$ is a vector of elements of type \texttt{DOA}. 
+
+The $\function(v)$ has type \texttt{Vec[DOA]} follows from the assumption that element $v$ is in \texttt{input\_domain} and from the type signature of \texttt{function} in line~\ref{line:fn} of the pseudocode (Section~\ref{sec:pseudocode}), which takes in an element of type \texttt{Vec(DIA)} and returns an element of type \texttt{Vec[DOA]}. If the Rust code compiles correctly, then the type correctness follows from the definition of the type signature enforced by Rust. Otherwise, the code raises a compile time error for incorrect function input type or output type. 
+
+Notice also that the appropriate output domain is achieved during run time since the \texttt{atom\_function} is a pure function with a function trait that maps elements of type \texttt{\&DIA::Carrier} to \texttt{DOA::Carrier}. Therefore the  \texttt{function} correctly maps elements in the input domain \texttt{VectorDomain(DIA)} to the output domain of \texttt{VectorDomain(DOA)} during run time.
+
+
+\item \textbf{(Domain-metric compatibility).} The Symmetric distance is both the \texttt{input\_metric} and \texttt{output\_metric}. Symmetric distance is compatible with \texttt{VectorDomain(T)} for any generic type \texttt{T}, as stated in \href{https://www.overleaf.com/project/60d215bf90b337ac02200a99}{``List of definitions used in the pseudocode"}. The theorem holds because for \texttt{make\_row\_by\_row}, the input domain is \texttt{VectorDomain(DIA)} and the output domain is \texttt{VectorDomain(DOA)}. 
```

Comment:
"for any generic type T" or "for any generic type T with the Domain trait"?

Reply:
Yes. It would also be clearer to use D instead of T when referring to a generic that is a domain.